### DISCUSSION TITLE
Build elementary OS under macOS

### DISCUSSION THREAD

**Discussion by srbala** [elementary/os#513] [Q&A] [answered]
Team, is possibe to build elementary iso in MacOS?

**Comment by cassidyjames** [comment]
@srbala I haven't personally tried it, but it happens in a Docker container so it _should_ work so long as you have Docker installed and configured for your user on macOS.

**Reply by srbala** [reply]
I was under the same impression. Ran the docker command from `64-bit AMD/Intel` secion. One of the issue is mac does not have '/proc' folder to mount. Following is partial log from the end of failere. Would need full log file?

```sh
#------------------#
# LIVE-BUILD BUILD #
#------------------#

[2021-08-03 16:45:13] lb build 
P: live-build 1:20190311
P: Building config tree for a ubuntu/bionic/amd64 system
[2021-08-03 16:45:13] lb bootstrap 
P: Setting up cleanup function
[2021-08-03 16:45:14] lb bootstrap_cache restore
[2021-08-03 16:45:14] lb bootstrap_debootstrap 
P: Begin bootstrapping system...
P: If the following stage fails, the most likely cause of the problem is with your mirror configuration or a caching proxy.
P: Running debootstrap... 
mknod: /working_dir/tmp/amd64/chroot/test-dev-null: Operation not permitted
E: Cannot install into target '/working_dir/tmp/amd64/chroot' mounted with noexec or nodev
P: Begin unmounting filesystems...
P: Saving caches...
```

**Reply by srbala** [reply]
@cassidyjames Can we use linux vms to build? which flavour of VM you would recommend (debian/centos etc)?

**Reply by cassidyjames** [reply]
@srbala sorry, building on macOS is out of my depth; I've just been building locally on elementary OS as needed for testing. I would think an elementary OS 6 or Ubuntu 20.04 VM would give you the closest experience to what we have when building on elementary OS, though.

**Comment by ryonakano** [comment, marked as answer]
Closing as we don't support building on macOS because we now require systemd to build.
